brave/brave-core
# Add a profile attribute migration for the avatar icon

Fixes https://github.com/brave/brave-browser/issues/40005

Brave originally allowed folks to pick the Chromium profile icons.
We then removed those in favor of our own branded icons in 0.70.x (2019).
https://github.com/brave/brave-core/pull/3165

The old ones would continue to work - but may have had rendering issues.
Chromium 127 had a Windows change which now triggers a CHECK.
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/profiles/profile_attributes_entry.cc;l=389-397;drc=b63e5a1c552b81ae2f83ce7684eb92edb711da1e

This migration moves folks who have the old IDs to the default profile ID.

Special thanks to:
- @emerick for helping me find the solution 
- @simonhong for helping me with the tests




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Manual test
1. Fresh profile
2. Exit Brave
3. Use an editor to open the `Local State` file at the root of the profile
4. Find the section `avatar_icon":"chrome://theme/IDR_PROFILE_`
5. Change the value of `avatar_icon` to be `chrome://theme/IDR_PROFILE_AVATAR_28` and save the file
6. Launch Brave
7. Close Brave
8. Use an editor to open the `Local State` file at the root of the profile
9. Find the section `avatar_icon":"chrome://theme/IDR_PROFILE_`
10. It should be set to `chrome://theme/IDR_PROFILE_AVATAR_26` now

### Automated test
Unit tests can be ran with:
```
npm run test -- brave_unit_tests --filter="ProfileAttributeMigrationTest.*"
```
